packages/cli/src/commands/build.ts

```diff
@@ -312,6 +313,43 @@ export default async function main(client: Client): Promise<number> {
   // TODO: parallelize builds
   const buildResults: Map<Builder, BuildResult> = new Map();
   const overrides: PathOverride[] = [];
+  const repoRootPath = cwd === workPath ? undefined : cwd;
+  const rootPackageJsonPath = repoRootPath || workPath;
+
+  if (process.env.ENABLE_EXPERIMENTAL_COREPACK === '1') {
+    const pkg = await readJSONFile<PackageJson>(
+      join(rootPackageJsonPath, 'package.json')
+    );
+    if (pkg instanceof CantParseJSONFile) {
+      console.warn(
+        'Warning: Could not enable corepack because package.json is invalid JSON'
+      );
+    } else if (!pkg?.packageManager) {
+      console.warn(
+        'Warning: Could not enable corepack because package.json is missing "packageManager" property'
+      );
+    } else {
+      console.log(
+        `Detected ENABLE_EXPERIMENTAL_COREPACK=1 and "${pkg.packageManager}" in package.json`
+      );
+      const cacheDir = join(
+        rootPackageJsonPath,
+        '.vercel',
+        'cache',
+        'corepack'
+      );
```

Comment:
@TooTallNate Do you have a preference for cache path here?

We'll need to add it to `prepareCache()` and I think we talked about adding `.vercel/cache/**` to always cache.

Thoughts?

Reply:
Seems good to me 👍

Reply:
Ok cool, I'll add `.vercel/cache/**` to the default build cache and then we can reuse this for new features going forward 👍